Core/include/Acts/Utilities/Range1D.hpp

```diff
@@ -0,0 +1,322 @@
+// This file is part of the Acts project.
+//
+// Copyright (C) 2016-2018 CERN for the benefit of the Acts project
+//
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+#pragma once
+
+#include <algorithm>
+#include <limits>
+
+namespace Acts {
+/// @brief A one-dimensional range between two points.
+///
+/// This type describes a one-demensional range of values, designed to be used
+/// in the construction of more complex multi-dimensional types. The class
+/// provides functionality for growing and shrinking the range, as well as some
+/// other utilities.
+///
+/// @tparam Type The scalar type of the values contained in this range.
+template <typename Type>
+class Range1D {
+ public:
+  /// @brief Construct a new degenerate range object
+  ///
+  /// This constructor coonstructs a degenerate range object with a maximum
+  /// lower than the minimum. In other words, this range is empty.
+  Range1D()
+      : m_min(std::numeric_limits<Type>::lowest()),
+        m_max(std::numeric_limits<Type>::max()) {}
+
+  /// @brief Construct a new range object from a lower and upper bound
+  ///
+  /// Construct a new range object given the values for the minimum and
+  /// maximum. Note that it is perfectly possible to construct a degenerate
+  /// range in this way.
+  ///
+  /// @param min The minimum value in the range (inclusive)
+  /// @param max The maximum value in the range (inclusive)
+  Range1D(Type min, Type max) : m_min(min), m_max(max) {}
```

Comment:
If you have a chain of Ranges, and both borders are inclusive, isn't that somewhat inconsistent?

Sure, that's just a numerical edge effect, but I was wondering if one could use a [x,y) range definition instead.

Reply:
Agreed, let's move to a half-open interval definition instead.